fastf1/core.py

```diff
@@ -1884,16 +1885,42 @@ def _check_lap_accuracy(self):
                                           atol=0.003, rtol=0, equal_nan=False)
                     if not check_2:
                         integrity_errors += 1
+                        lap_integrity_flag = False
                 else:
                     check_2 = False  # data not available means fail
 
                 if prev_lap is not None:
+                    if (not pd.isnull(prev_lap['Time'])):
+                        prev_lap_check = True  # it's needed for check_4
+                    else:
+                        prev_lap_check = False
                     # first lap after safety car often has timing issues (as do all laps under safety car)
                     check_3 = (prev_lap['TrackStatus'] != '4')
                 else:
+                    prev_lap_check = False
                     check_3 = True  # no previous lap, no SC error
 
-                result = check_1 and check_2 and check_3
+                pre_check_4 = (prev_lap_check
+                               & (not pd.isnull(lap['Time']))
+                               & (not pd.isnull(lap['LapTime'])))
+
+                if pre_check_4:  # needed condition for check_4
```

Comment:
Instead of setting prev_lap_check above, just do
``if pre_check_4 and (prev_lap is not None) and (not pd.isnull(prev_lap['Time']))``